src/tpm2-tss-engine.c

```diff
@@ -270,7 +271,9 @@ init_engine(ENGINE *e) {
 #ifdef ENABLE_TCTIENVVAR
     /*  Set the default TCTI option from the environment */
     OPENSSL_free(tcti_nameconf);
-    tcti_nameconf = OPENSSL_strdup(getenv("TPM2TSSENGINE_TCTI"));
+    if (getenv("TPM2TSSENGINE_TCTI")) {
+        tcti_nameconf = OPENSSL_strdup(getenv("TPM2TSSENGINE_TCTI"));
+    }
```

Comment:
This hunk here is pretty important, it segfaults if it's ENABLE_TCTIENVVAR is defined but the env var TPM2TSSENGINE_TCTI is not set.... I can split this out to a separate PR if you want to take it but need changes elsewhere in this PR

Reply:
The reason, at least as seen in Ubuntu 16.04 and its version of OSSL is that OPENSSL_strdup tries to access the NULL pointer.